app/features/send/cashu-send-quote-hooks.ts

```diff
@@ -0,0 +1,500 @@
+        (sendQuote) => sendQuote.quoteId === meltQuote.quote,
+      );
+
+      if (!relatedSendQuote) {
+        console.warn('No related send quote found for the melt quote');
+        return;
+      }
+
+      const account = await accountsCache.getLatest(relatedSendQuote.accountId);
+      if (!account || account.type !== 'cashu') {
+        throw new Error(
+          `Account not found for id: ${relatedSendQuote.accountId}`,
+        );
+      }
+
+      const expiresAt = new Date(relatedSendQuote.expiresAt);
+      const now = new Date();
+
+      if (
+        meltQuote.state === 'UNPAID' &&
+        expiresAt < now &&
+        relatedSendQuote.state !== 'EXPIRED'
+      ) {
+        // TODO: do we get notified by the socket when the quote is expired?
+        // If not we need to find another way.
+        onExpiredRef.current(account, relatedSendQuote, meltQuote);
+      } else if (
+        meltQuote.state === 'PAID' &&
+        relatedSendQuote.state !== 'PAID'
+      ) {
+        onPaidRef.current(account, relatedSendQuote, meltQuote);
+      } else if (
+        meltQuote.state === 'PENDING' &&
+        relatedSendQuote.state !== 'PENDING'
+      ) {
+        onPendingRef.current(account, relatedSendQuote, meltQuote);
+      } else if (
+        meltQuote.state === 'UNPAID' &&
+        relatedSendQuote.state === 'UNPAID'
+      ) {
+        onUnpaidRef.current(account, relatedSendQuote, meltQuote);
+      }
+    },
+    [sendQuotes, accountsCache],
+  );
+
+  useEffect(() => {
+    if (sendQuotes.length === 0) return;
+
+    const subcribeToMeltQuoteUpdates = async (quotes: CashuSendQuote[]) => {
+      const quotesByMint = quotes.reduce<Record<string, string[]>>(
+        (acc, quote) => {
+          const account = accountsCache.get(quote.accountId);
+          if (!account || account.type !== 'cashu') {
+            throw new Error(
+              `Cashu account not found for id: ${quote.accountId}`,
+            );
+          }
+          const existingQuotesForMint = acc[account.mintUrl] ?? [];
+          acc[account.mintUrl] = existingQuotesForMint.concat(quote.quoteId);
+          return acc;
+        },
+        {},
+      );
+
+      const subscriptions = Object.entries(quotesByMint).map(
+        ([mintUrl, quotes]) => {
+          const wallet = getCashuWallet(mintUrl);
+          return wallet.onMeltQuoteUpdates(
+            quotes,
+            handleMeltQuoteUpdate,
+            // TODO: what should we do here on error?
+            (error) =>
+              console.error('Melt quote updates socket error', {
+                cause: error,
+              }),
+          );
+        },
+      );
+
+      const subscriptionCancellers = await Promise.all(subscriptions);
+
+      return () => {
+        subscriptionCancellers.forEach((cancel) => cancel());
+      };
+    };
+
+    subcribeToMeltQuoteUpdates(sendQuotes);
+  }, [sendQuotes, accountsCache, handleMeltQuoteUpdate]);
+
+  useEffect(() => {
```

Comment:
added this useEffect to handle meltQuote expiration because if it has expired without the state change the socket won't notify us so instead I scheduled a check at expiration time